Merge course scoresheet and series status overview

LucianosLionakis opened the pull request:
Fixes #1075

MrKickkiller commented on app/models/course.rb:
```diff
@@ -205,7 +205,9 @@ def scoresheet(options = {})
         row = [user.first_name, user.last_name, user.username, user.email]
         sorted_series.each do |s|
-          row << s.exercises.map { |ex| ex.accepted_for(user, s.deadline, self) }.count(true)
+          latest_subs = Submission.where(user_id: user.id, course_id: id, exercise_id: s.exercises.map(&:id)).select('MAX(id) as id')
```
Niet getest, maar zou het volgende niet mooier zijn dan eerst te mappen en dan te selecten?
```
s.exercises.maximum(:id)
```
https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-maximum

winniederidder replied:
De select max(id) slaat op de Submission id van de gevonden Submissions, dus niet van de exercises. De select as id is nodig opdat ik daarna where(id: ...) kan uitvoeren. Toch bedankt.

charvp commented on app/assets/javascripts/index.js:
```diff
@@ -218,7 +219,9 @@ function initFilterIndex(_baseUrl, eager, actions, doInitFilter, filterCollectio
                 const $link = $(
-                    `<a class="action" href='#' ${
+                    `<a class="action" href=${
+                        action.url ? action.url : "#"
```
`action.url` won't have quotes around it here.

charvp commented on app/assets/javascripts/index.js:
```diff
@@ -229,7 +232,7 @@ function initFilterIndex(_baseUrl, eager, actions, doInitFilter, filterCollectio
                         return false;
                     });
-                } else {
+                } else if (action.js) {
```
Why this extra `if`?

LucianosLionakis replied:
Because the link for downloading the scoresheet doesn't work without the extra if. I think it is because an on click event is made with action.js (which isn't defined).

charvp commented on app/assets/javascripts/index.js:
```diff
@@ -155,15 +155,15 @@ function initFilterIndex(_baseUrl, eager, actions, doInitFilter, filterCollectio
-                        .parent()
-                        .hasClass("active")
+                    $($(`a.action[data-search_opt_id="${id}"]`)
+                        .children()[0])
```
I would prefer an explicit `find`. It gets rid of the double jQuery call and can probably withstand changes in the markup better, should they occur.

LucianosLionakis commented:
Moved the checkbox after the text. It now looks like this:
![DeepinScreenshot_select-area_20190909104944](https://user-images.githubusercontent.com/48768719/64516938-9e845480-d2ef-11e9-8d25-336b974298a5.png)

I changed the dropdown paddings so that there is enough place for the checkbox

charvp commented on app/assets/javascripts/index.js:
```diff
@@ -192,7 +192,7 @@ function initFilterIndex(_baseUrl, eager, actions, doInitFilter, filterCollectio
                     `<a class="action" href='#' ${
                         action.type ? "data-type=" + action.type : ""
-                    } data-search_opt_id="${id}"><i class='mdi mdi-checkbox-blank-outline mdi-18'></i>${
+                    } data-search_opt_id="${id}"><i class='mdi mdi-checkbox-blank-outline mdi-18 mdi-box'></i>${
```
Can't you just place the icon after `action.text`?

charvp commented on vendor/assets/stylesheets/bootstrap/dropdowns.less:
```diff
@@ -64,7 +64,7 @@
   > li > a {
     display: block;
-    padding: 3px 20px;
+    padding: 3px 30px 3px 20px;
```
This definitely shouldn't be changed for all dropdowns. (Editing things in the vendor directory shouldn't be done either way.)

charvp commented on app/assets/stylesheets/material_icons.css.less:
```diff
@@ -74,6 +74,10 @@ a.btn-fab .mdi::before{
 }
+.dropdown-menu .mdi-box::before {
+  left: 10px;
+  position: relative; // required to use top and left
```
Update the comment
